### PR TITLE
Add option to set proxy settings

### LINKED ISSUE #235: Allow users to specify/bypass HTTP proxies for local traffic to ycmd
We've seen cases where HTTP proxies are managing the traffic to ycmd, and in some cases this means it's sent through a firewall which block it. See for example #232.

In #232 @Alexander-Shukaev has some good suggestions on how we can approach this. The basic idea is to use a `let`to temporarily rebind `url-proxy-services` around our calls which generate HTTP traffic. Should be very straightforward.

### PR DESCRIPTION
**Pull request by ptrv**
Allow to bypass proxy settings for the communication with the ycmd
server.

Resolves #235

### CODE REVIEW

File: ycmd.el [hunk 1]
```diff
@@ -254,6 +254,26 @@ use `ycmd-parse-buffer'."
...
+Sets the variable `url-proxy-services' in `ycmd--request' when
+sending requests to the ycmd server:
+
+nil
+    Do not use a proxy server at all.
+
+`default'
+    Use the value of `url-proxy-services'.
+
+Alist
```
Comment (ptrv) [inline ycmd.el]: Do we even need this? I think `nil` or the value of `url-proxy-settings` should be enough.

Comment (abingham) [inline ycmd.el]: Honestly I think a bypass boolean is the way to go right now. The only use case we have for this feature is bypassing proxies, so the ability to specify a completely alternative list feels like feature bloat. So probably something like `ycmd-bypass-url-proxy-services` which defaults to true.

**Comment by ptrv** [comment]
@abingham Updated the PR. We use a bool now to bypass `url-proxy-services`.

**Comment by abingham** [comment]
This looks good! I'd like the original reporter to test this out just for completeness.

@Alexander-Shukaev Can you try this out just so we can be sure it addresses the initial issue?

**Comment by Alexander-Shukaev** [comment]
Works just fine.  Thank you.  I just want to point out that with this approach, users won't be able to set a dedicated proxy for YCMD, if that would be the case.  That is my proposal for `ycmd-url-proxy-services` is more generic.  Nevertheless, I do admit that for the vast majority of use cases (including my current ones) this fix is sufficient.  Looking forward to see it merged.

**Comment by ptrv** [comment]
@Alexander-Shukaev Thanks for testing!

**Pull request merged**